### PR TITLE
Replace Composition* pkgprojs with NuGet Pack task

### PR DESCRIPTION
**Pull request by ViktorHofer**
System.Composition is a meta-package only so that required a bit try and
error but that worked out quite well. The added dependency groups in
that meta-package are intentional.

In addition to that, little cleanup:
1. Moving the PackageDescription from PerformanceCounter's Directory.Build.props file into its src csproj file.
2. Moving the StrongNameKey property from the Composition projects' Directory.Build.props file into their csproj as they don't have a ref project so the StrongNameKey doesn't need to be shared.

TODO:
- Share the nuspec diff

Contributes to https://github.com/dotnet/runtime/issues/53202

### CODE REVIEW

File: src/libraries/System.Composition.TypedParts/src/CompatibilitySuppressions.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0"?>
+<Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+  <!-- .NETStandard,Version=v1.0 and .NETPortable,Version=v0.0,Profile=Profile259 are intentionally dropped from the package. -->
+  <Suppression>
```
Comment (Anipik) [inline src/libraries/System.Composition.TypedParts/src/CompatibilitySuppressions.xml:4]: it seems like suppressions file is same for all the projects. can we have just one copy ?

**Review by Anipik — COMMENTED** [review]
LGTM, just a single comment.
You can get the diff or try running this https://github.com/Anipik/PackagePortTester on the nupkgs

File: src/libraries/System.Composition.TypedParts/src/CompatibilitySuppressions.xml (hunk 1 above)
Comment (ViktorHofer) [reply src/libraries/System.Composition.TypedParts/src/CompatibilitySuppressions.xml:4]: Considered that but as this is just temporary until we ship the NET6 packages I would prefer to keep the suppression files for each project and then clean them all together up in NET7.

**Comment by ViktorHofer** [comment]
Here's the diff:

```diff
 <?xml version="1.0" encoding="utf-8"?>
-<package xmlns="http://schemas.microsoft.com/packaging/2013/01/nuspec.xsd">
-  <metadata minClientVersion="2.12">
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
     <id>System.Composition</id>
-    <version>6.0.0-preview.5.21301.5</version>
-    <title>System.Composition</title>
+    <version>6.0.0-dev</version>
     <authors>Microsoft</authors>
-    <owners>microsoft,dotnetframework</owners>
+    <owners></owners>
     <requireLicenseAcceptance>false</requireLicenseAcceptance>
     <license type="expression">MIT</license>
     <icon>Icon.png</icon>
     <projectUrl>https://dot.net/</projectUrl>
-    <iconUrl>http://go.microsoft.com/fwlink/?LinkID=288859</iconUrl>
     <description>This packages provides a version of the Managed Extensibility Framework (MEF) that is lightweight and specifically optimized for high throughput scenarios, such as the web.
 
 Commonly Used Types:
@@ -23,29 +21,43 @@ System.Composition.CompositionContextExtensions</description>
     <releaseNotes>https://go.microsoft.com/fwlink/?LinkID=799421</releaseNotes>
     <copyright>© Microsoft Corporation. All rights reserved.</copyright>
     <serviceable>true</serviceable>
-    <repository type="git" url="https://github.com/dotnet/runtime" commit="ec3e0b276b159ab24946f17f2b89d00862488a6a" />
+    <repository type="git" url="https://github.com/dotnet/runtime" commit="0000000000000000000000000000000000000000" />
     <dependencies>
-      <dependency id="System.Composition.AttributedModel" version="6.0.0-preview.5.21301.5" />
-      <dependency id="System.Composition.Convention" version="6.0.0-preview.5.21301.5" />
-      <dependency id="System.Composition.Hosting" version="6.0.0-preview.5.21301.5" />
-      <dependency id="System.Composition.Runtime" version="6.0.0-preview.5.21301.5" />
-      <dependency id="System.Composition.TypedParts" version="6.0.0-preview.5.21301.5" />
+      <group targetFramework=".NETFramework4.6.1">
+        <dependency id="System.Composition.AttributedModel" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.Convention" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.Hosting" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.Runtime" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.TypedParts" version="6.0.0-dev" exclude="Build,Analyzers" />
+      </group>
+      <group targetFramework="net6.0">
+        <dependency id="System.Composition.AttributedModel" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.Convention" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.Hosting" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.Runtime" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.TypedParts" version="6.0.0-dev" exclude="Build,Analyzers" />
+      </group>
+      <group targetFramework=".NETStandard2.0">
+        <dependency id="System.Composition.AttributedModel" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.Convention" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.Hosting" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.Runtime" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.TypedParts" version="6.0.0-dev" exclude="Build,Analyzers" />
+      </group>
     </dependencies>
   </metadata>
 </package>
 <?xml version="1.0" encoding="utf-8"?>
-<package xmlns="http://schemas.microsoft.com/packaging/2013/01/nuspec.xsd">
-  <metadata minClientVersion="2.12">
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
     <id>System.Composition.AttributedModel</id>
-    <version>6.0.0-preview.5.21301.5</version>
-    <title>System.Composition.AttributedModel</title>
+    <version>6.0.0-dev</version>
     <authors>Microsoft</authors>
-    <owners>microsoft,dotnetframework</owners>
+    <owners></owners>
     <requireLicenseAcceptance>false</requireLicenseAcceptance>
     <license type="expression">MIT</license>
     <icon>Icon.png</icon>
     <projectUrl>https://dot.net/</projectUrl>
-    <iconUrl>http://go.microsoft.com/fwlink/?LinkID=288859</iconUrl>
     <description>Provides common attributes used by System.Composition types.
 
 Commonly Used Types:
@@ -55,28 +67,24 @@ System.Composition.Convention.AttributedModelProvider</description>
     <releaseNotes>https://go.microsoft.com/fwlink/?LinkID=799421</releaseNotes>
     <copyright>© Microsoft Corporation. All rights reserved.</copyright>
     <serviceable>true</serviceable>
-    <repository type="git" url="https://github.com/dotnet/runtime" commit="ec3e0b276b159ab24946f17f2b89d00862488a6a" />
+    <repository type="git" url="https://github.com/dotnet/runtime" commit="0000000000000000000000000000000000000000" />
     <dependencies>
       <group targetFramework=".NETFramework4.6.1" />
+      <group targetFramework="net6.0" />
       <group targetFramework=".NETStandard2.0" />
     </dependencies>
-    <frameworkAssemblies>
-      <frameworkAssembly assemblyName="mscorlib" targetFramework=".NETFramework4.6.1" />
-    </frameworkAssemblies>
   </metadata>
 </package><?xml version="1.0" encoding="utf-8"?>
-<package xmlns="http://schemas.microsoft.com/packaging/2013/01/nuspec.xsd">
-  <metadata minClientVersion="2.12">
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
     <id>System.Composition.Convention</id>
-    <version>6.0.0-preview.5.21301.5</version>
-    <title>System.Composition.Convention</title>
+    <version>6.0.0-dev</version>
     <authors>Microsoft</authors>
-    <owners>microsoft,dotnetframework</owners>
+    <owners></owners>
     <requireLicenseAcceptance>false</requireLicenseAcceptance>
     <license type="expression">MIT</license>
     <icon>Icon.png</icon>
     <projectUrl>https://dot.net/</projectUrl>
-    <iconUrl>http://go.microsoft.com/fwlink/?LinkID=288859</iconUrl>
     <description>Provides types that support using Managed Extensibility Framework with a convention-based configuration model.
 
 Commonly Used Types:
@@ -88,34 +96,31 @@ System.Composition.Convention.ParameterImportConventionBuilder</description>
     <releaseNotes>https://go.microsoft.com/fwlink/?LinkID=799421</releaseNotes>
     <copyright>© Microsoft Corporation. All rights reserved.</copyright>
     <serviceable>true</serviceable>
-    <repository type="git" url="https://github.com/dotnet/runtime" commit="ec3e0b276b159ab24946f17f2b89d00862488a6a" />
+    <repository type="git" url="https://github.com/dotnet/runtime" commit="0000000000000000000000000000000000000000" />
     <dependencies>
       <group targetFramework=".NETFramework4.6.1">
-        <dependency id="System.Composition.AttributedModel" version="6.0.0-preview.5.21301.5" />
+        <dependency id="System.Composition.AttributedModel" version="6.0.0-dev" exclude="Build,Analyzers" />
+      </group>
+      <group targetFramework="net6.0">
+        <dependency id="System.Composition.AttributedModel" version="6.0.0-dev" exclude="Build,Analyzers" />
       </group>
       <group targetFramework=".NETStandard2.0">
-        <dependency id="System.Composition.AttributedModel" version="6.0.0-preview.5.21301.5" />
+        <dependency id="System.Composition.AttributedModel" version="6.0.0-dev" exclude="Build,Analyzers" />
       </group>
     </dependencies>
-    <frameworkAssemblies>
-      <frameworkAssembly assemblyName="mscorlib" targetFramework=".NETFramework4.6.1" />
-      <frameworkAssembly assemblyName="System.Core" targetFramework=".NETFramework4.6.1" />
-    </frameworkAssemblies>
   </metadata>
 </package>
 <?xml version="1.0" encoding="utf-8"?>
-<package xmlns="http://schemas.microsoft.com/packaging/2013/01/nuspec.xsd">
-  <metadata minClientVersion="2.12">
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
     <id>System.Composition.Hosting</id>
-    <version>6.0.0-preview.5.21301.5</version>
-    <title>System.Composition.Hosting</title>
+    <version>6.0.0-dev</version>
     <authors>Microsoft</authors>
-    <owners>microsoft,dotnetframework</owners>
+    <owners></owners>
     <requireLicenseAcceptance>false</requireLicenseAcceptance>
     <license type="expression">MIT</license>
     <icon>Icon.png</icon>
     <projectUrl>https://dot.net/</projectUrl>
-    <iconUrl>http://go.microsoft.com/fwlink/?LinkID=288859</iconUrl>
     <description>Provides Managed Extensibility Framework types that are useful to developers of extensible applications, or hosts.
 
 Commonly Used Types:
@@ -123,36 +128,34 @@ System.Composition.Hosting.CompositionHost</description>
     <releaseNotes>https://go.microsoft.com/fwlink/?LinkID=799421</releaseNotes>
     <copyright>© Microsoft Corporation. All rights reserved.</copyright>
     <serviceable>true</serviceable>
-    <repository type="git" url="https://github.com/dotnet/runtime" commit="ec3e0b276b159ab24946f17f2b89d00862488a6a" />
+    <repository type="git" url="https://github.com/dotnet/runtime" commit="0000000000000000000000000000000000000000" />
     <dependencies>
       <group targetFramework=".NETFramework4.6.1">
-        <dependency id="System.Composition.Runtime" version="6.0.0-preview.5.21301.5" />
+        <dependency id="System.Composition.Runtime" version="6.0.0-dev" exclude="Build,Analyzers" />
+      </group>
+      <group targetFramework="net6.0">
+        <dependency id="System.Composition.Runtime" version="6.0.0-dev" exclude="Build,Analyzers" />
       </group>
       <group targetFramework=".NETStandard2.0">
-        <dependency id="System.Composition.Runtime" version="6.0.0-preview.5.21301.5" />
+        <dependency id="System.Composition.Runtime" version="6.0.0-dev" exclude="Build,Analyzers" />
       </group>
     </dependencies>
     <frameworkAssemblies>
-      <frameworkAssembly assemblyName="mscorlib" targetFramework=".NETFramework4.6.1" />
-      <frameworkAssembly assemblyName="System" targetFramework=".NETFramework4.6.1" />
       <frameworkAssembly assemblyName="System.ComponentModel.Composition" targetFramework=".NETFramework4.6.1" />
-      <frameworkAssembly assemblyName="System.Core" targetFramework=".NETFramework4.6.1" />
     </frameworkAssemblies>
   </metadata>
 </package>
 <?xml version="1.0" encoding="utf-8"?>
-<package xmlns="http://schemas.microsoft.com/packaging/2013/01/nuspec.xsd">
-  <metadata minClientVersion="2.12">
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
     <id>System.Composition.Runtime</id>
-    <version>6.0.0-preview.5.21301.5</version>
-    <title>System.Composition.Runtime</title>
+    <version>6.0.0-dev</version>
     <authors>Microsoft</authors>
-    <owners>microsoft,dotnetframework</owners>
+    <owners></owners>
     <requireLicenseAcceptance>false</requireLicenseAcceptance>
     <license type="expression">MIT</license>
     <icon>Icon.png</icon>
     <projectUrl>https://dot.net/</projectUrl>
-    <iconUrl>http://go.microsoft.com/fwlink/?LinkID=288859</iconUrl>
     <description>Contains runtime components of the Managed Extensibility Framework.
 
 Commonly Used Types:
@@ -160,30 +163,25 @@ System.Composition.CompositionContext</description>
     <releaseNotes>https://go.microsoft.com/fwlink/?LinkID=799421</releaseNotes>
     <copyright>© Microsoft Corporation. All rights reserved.</copyright>
     <serviceable>true</serviceable>
-    <repository type="git" url="https://github.com/dotnet/runtime" commit="ec3e0b276b159ab24946f17f2b89d00862488a6a" />
+    <repository type="git" url="https://github.com/dotnet/runtime" commit="0000000000000000000000000000000000000000" />
     <dependencies>
       <group targetFramework=".NETFramework4.6.1" />
+      <group targetFramework="net6.0" />
       <group targetFramework=".NETStandard2.0" />
     </dependencies>
-    <frameworkAssemblies>
-      <frameworkAssembly assemblyName="mscorlib" targetFramework=".NETFramework4.6.1" />
-      <frameworkAssembly assemblyName="System.Core" targetFramework=".NETFramework4.6.1" />
-    </frameworkAssemblies>
   </metadata>
 </package>
 <?xml version="1.0" encoding="utf-8"?>
-<package xmlns="http://schemas.microsoft.com/packaging/2013/01/nuspec.xsd">
-  <metadata minClientVersion="2.12">
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
     <id>System.Composition.TypedParts</id>
-    <version>6.0.0-preview.5.21301.5</version>
-    <title>System.Composition.TypedParts</title>
+    <version>6.0.0-dev</version>
     <authors>Microsoft</authors>
-    <owners>microsoft,dotnetframework</owners>
+    <owners></owners>
     <requireLicenseAcceptance>false</requireLicenseAcceptance>
     <license type="expression">MIT</license>
     <icon>Icon.png</icon>
     <projectUrl>https://dot.net/</projectUrl>
-    <iconUrl>http://go.microsoft.com/fwlink/?LinkID=288859</iconUrl>
     <description>Provides some extension methods for the Managed Extensibility Framework.
 
 Commonly Used Types:
@@ -192,22 +190,23 @@ System.Composition.Hosting.ContainerConfiguration</description>
     <releaseNotes>https://go.microsoft.com/fwlink/?LinkID=799421</releaseNotes>
     <copyright>© Microsoft Corporation. All rights reserved.</copyright>
     <serviceable>true</serviceable>
-    <repository type="git" url="https://github.com/dotnet/runtime" commit="ec3e0b276b159ab24946f17f2b89d00862488a6a" />
+    <repository type="git" url="https://github.com/dotnet/runtime" commit="0000000000000000000000000000000000000000" />
     <dependencies>
       <group targetFramework=".NETFramework4.6.1">
-        <dependency id="System.Composition.AttributedModel" version="6.0.0-preview.5.21301.5" />
-        <dependency id="System.Composition.Hosting" version="6.0.0-preview.5.21301.5" />
-        <dependency id="System.Composition.Runtime" version="6.0.0-preview.5.21301.5" />
+        <dependency id="System.Composition.AttributedModel" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.Hosting" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.Runtime" version="6.0.0-dev" exclude="Build,Analyzers" />
+      </group>
+      <group targetFramework="net6.0">
+        <dependency id="System.Composition.AttributedModel" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.Hosting" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.Runtime" version="6.0.0-dev" exclude="Build,Analyzers" />
       </group>
       <group targetFramework=".NETStandard2.0">
-        <dependency id="System.Composition.AttributedModel" version="6.0.0-preview.5.21301.5" />
-        <dependency id="System.Composition.Hosting" version="6.0.0-preview.5.21301.5" />
-        <dependency id="System.Composition.Runtime" version="6.0.0-preview.5.21301.5" />
+        <dependency id="System.Composition.AttributedModel" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.Hosting" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="System.Composition.Runtime" version="6.0.0-dev" exclude="Build,Analyzers" />
       </group>
     </dependencies>
-    <frameworkAssemblies>
-      <frameworkAssembly assemblyName="mscorlib" targetFramework=".NETFramework4.6.1" />
-      <frameworkAssembly assemblyName="System.Core" targetFramework=".NETFramework4.6.1" />
-    </frameworkAssemblies>
   </metadata>
 </package>

```

**Review by Anipik — APPROVED** [review]

**Comment by ViktorHofer** [comment]
CoreClr failure is an ongoing AzDO feed restore issue.

**Pull request merged**